spec.md

```diff
@@ -575,23 +597,28 @@ message CreateVolumeRequest {
   map<string, string> parameters = 4;
 
   // Secrets required by plugin to complete volume creation request.
-  // A secret is a string to string map where the key identifies the
-  // name of the secret (e.g. "username" or "password"), and the value
-  // contains the secret data (e.g. "bob" or "abc123").
-  // Each key MUST consist of alphanumeric characters, '-', '_' or '.'.
-  // Each value MUST contain a valid string. An SP MAY choose to accept
-  // binary (non-string) data by using a binary-to-text encoding scheme,
-  // like base64.
-  // An SP SHALL advertise the requirements for required secret keys and
-  // values in documentation.
-  // CO SHALL permit passing through the required secrets.
-  // A CO MAY pass the same secrets to all RPCs, therefore the keys for
-  // all unique secrets that an SP expects must be unique across all CSI
-  // operations.
-  // This information is sensitive and MUST be treated as such (not
-  // logged, etc.) by the CO.
   // This field is OPTIONAL.
   map<string, string> controller_create_secrets = 5;
+
+  // Contains all attributes of the source that is used to create
+  // the volume from. This field is OPTIONAL.
+  VolumeContentSource content_source = 6;
+}
+
+// Specifies what source the volume will be created from. One of the
+// source_type fields MUST be specified. Only Snapshot is
+// a supported VolumeContentSource at the moment.
+message VolumeContentSource {
+  message Snapshot {
```

Comment:
This is confusing, because there is already a top level (global) `message Snapshot`.

Maybe call it `SnapshotSource` instead of just `Snapshot`?

Reply:
Sure.